### PR TITLE
e2e - Prepare tests for independent work in threads run

### PR DESCRIPTION
**Pull request by tanyatrayanava**

### CODE REVIEW

File: tests/e2e/tests/smoke/browser/list-key.e2e.ts [hunk 1]
```diff
@@ -1,47 +1,38 @@
...
 const browserPage = new BrowserPage();
-const userAgreementPage = new UserAgreementPage();
-const addRedisDatabasePage = new AddRedisDatabasePage();
+const chance = new Chance();
+
+let keyName = chance.word({ length: 10 });
+const keyTTL = '2147476121';
+const element = '1111listElement11111';
+const element2 = '2222listElement22222';
+const element3 = '33333listElement33333';
```
Comment (elenaNaboko) [inline tests/e2e/tests/smoke/browser/list-key.e2e.ts:13]: The same: not random elements name

File: tests/e2e/tests/smoke/browser/list-of-keys-verifications.e2e.ts [hunk 2]
```diff
@@ -1,64 +1,70 @@
...
+        await t.click(browserPage.refreshKeysButton);
+        await t.expect(browserPage.keyNameInTheList.exists).ok('The list of keys is displayed');
+    });
+test
+    .after(async () => {
+        //Clear and delete database
+        await browserPage.deleteKeyByName(keyName1);
+        await browserPage.deleteKeyByName(keyName2);
+        await browserPage.deleteKeyByName(keyName3);
+        await browserPage.deleteKeyByName(keyName4);
```
Comment (elenaNaboko) [inline tests/e2e/tests/smoke/browser/list-of-keys-verifications.e2e.ts:53]: Maybe it would be better to perform deletion inside of the loop?

**Review by elenaNaboko — CHANGES_REQUESTED** [review]

File: tests/e2e/tests/critical-path/browser/context.e2e.ts [hunk 3]
```diff
@@ -92,8 +83,7 @@ test('Verify that user can see saved executed commands in CLI on Browser page wh
     }
 });
 test('Verify that user can see saved input entered into the filter per Key name when he returns back to Browser page', async t => {
-    const keyName = 'keyName123!.//';
-    await myRedisDatabasePage.clickOnDBByName(ossStandaloneConfig.databaseName);
+    const keyName = chance.word({ length: 10 });
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/context.e2e.ts:86]: You already defined it

File: tests/e2e/tests/critical-path/browser/database-overview.e2e.ts [hunk 4]
```diff
@@ -93,59 +97,86 @@ test('Verify that when user adds or deletes a new key, info in DB header is upda
...
+    })
+    ('Verify that user can see total number of keys rounded in format 100, 1K, 1M, 1B in DB header in Browser page', async t => {
+        //Add 100 keys
+        keys1 = await common.createArrayWithKeyValue(100);
+        await cliPage.sendCommandInCli(`MSET ${keys1.join(' ')}`);
+        //Wait 5 seconds
+        await t.wait(fiveSecondsTimeout);
+        //Verify that the info on DB header is updated after adds
+        let totalKeys = await browserPage.overviewTotalKeys.innerText;
+        await t.expect(totalKeys).eql('100', 'Info in DB header after ADD 100 keys');
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/database-overview.e2e.ts:118]: Seems you repeat this action in the test. Please create a function

File: tests/e2e/tests/critical-path/browser/filtering.e2e.ts [hunk 5]
```diff
@@ -1,69 +1,66 @@
...
+        await acceptLicenseTermsAndAddDatabase(ossStandaloneConfig, ossStandaloneConfig.databaseName);
+    })
+test
+    .after(async () => {
+        //Clear and delete database
+        await browserPage.deleteKeyByName(keyName);
+        await deleteDatabase(ossStandaloneConfig.databaseName);
+    })
+    ('Verify that user can search a key with selected data type is filters', async t => {
+        keyName = chance.word({ length: 10 });
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/filtering.e2e.ts:33]: I see it's already defined on line 18, why need to do that again here?

File: tests/e2e/tests/critical-path/browser/filtering.e2e.ts [hunk 6]
```diff
@@ -1,69 +1,66 @@
...
+import { Chance } from 'chance';
 
-const myRedisDatabasePage = new MyRedisDatabasePage();
 const browserPage = new BrowserPage();
-const userAgreementPage = new UserAgreementPage();
-const addRedisDatabasePage = new AddRedisDatabasePage();
 const cliPage = new CliPage();
+const chance = new Chance();
+
+let keyName = chance.word({ length: 10 });
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/filtering.e2e.ts:18]: Should be `const`. I didn't see you change it over the test

File: tests/e2e/tests/critical-path/browser/hash-field.e2e.ts [hunk 7]
```diff
@@ -1,39 +1,29 @@
...
+import { commonUrl, ossStandaloneConfig } from '../../../helpers/conf';
+import { Chance } from 'chance';
 
-const myRedisDatabasePage = new MyRedisDatabasePage();
 const browserPage = new BrowserPage();
-const userAgreementPage = new UserAgreementPage();
-const addRedisDatabasePage = new AddRedisDatabasePage();
+const chance = new Chance();
+
+let keyName = chance.word({ length: 10 });
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/hash-field.e2e.ts:9]: Same here, please fix that over all of the tests.

File: tests/e2e/tests/critical-path/browser/scan-keys.e2e.ts [hunk 8]
```diff
@@ -1,25 +1,25 @@
...
+import { Chance } from 'chance';
 
 const myRedisDatabasePage = new MyRedisDatabasePage();
 const browserPage = new BrowserPage();
-const userAgreementPage = new UserAgreementPage();
-const addRedisDatabasePage = new AddRedisDatabasePage();
 const settingsPage = new SettingsPage();
 const cliPage = new CliPage();
 const common = new Common();
+const chance = new Chance();
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/browser/scan-keys.e2e.ts:20]: I couldn't find usage in this test.

File: tests/e2e/tests/critical-path/cli/cli-critical.e2e.ts [hunk 9]
```diff
@@ -1,58 +1,59 @@
...
-const userAgreementPage = new UserAgreementPage();
-const addRedisDatabasePage = new AddRedisDatabasePage();
 const common = new Common();
+const browserPage = new BrowserPage();
+const chance = new Chance();
 
 const pairsToSet = common.createArrayPairsWithKeyValue(4);
 const MAX_AUTOCOMPLETE_EXECUTIONS = 100;
+let keyName = chance.word({ length: 10 });
+let value = chance.natural({ length: 5 });
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/cli/cli-critical.e2e.ts:23]: Should be const

**Review by ofersteinberg — CHANGES_REQUESTED** [review]
Next time, please open a smaller PR. It's rough to review such a big PR like that.

File: tests/e2e/tests/critical-path/workbench/scripting-area.e2e.ts [hunk 10]
```diff
@@ -84,74 +77,59 @@ test.skip('Verify that user when he have more than 10 results can request to vie
...
-        for(const commandPart of multipleLinesCommand) {
-            await t.expect(resultCommand).contains(commandPart, 'The multiple lines command is in the result');
-        }
-    });
+test.skip('Verify that user can see result in Table and Text views for Hash data types for FT.SEARCH command in Workbench', async t => {
+    indexName = chance.word({ length: 5 });
+    const commandsForSend = [
+        `FT.CREATE ${indexName} ON HASH PREFIX 1 product: SCHEMA name TEXT`,
+        'HMSET product:1 name "Apple Juice" ',
+        'HMSET product:2 name "Apple Juice"'
```
[the remark this answers is not in the document]
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/workbench/scripting-area.e2e.ts]: done

File: tests/e2e/tests/smoke/browser/list-of-keys-verifications.e2e.ts (hunk 2 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/smoke/browser/list-of-keys-verifications.e2e.ts]: done

File: tests/e2e/helpers/common.ts [hunk 11]
```diff
@@ -34,8 +36,8 @@ export class Common {
     async createArrayWithKeyValue(length: number): Promise<string[]> {
         const arr = [];
         for(let i = 1; i <= length * 2; i++) {
-            arr[i] = `key${i}`;
-            arr[i + 1] = `value${i}`;
+            arr[i] = `${chance.word({ length: 10 })}key${i}`;
+            arr[i + 1] = `${chance.word({ length: 10 })}value${i}`;
```
[the remark this answers is not in the document]
Comment (tanyatrayanava) [reply tests/e2e/helpers/common.ts]: done

File: tests/e2e/helpers/conf.ts [hunk 12]
```diff
@@ -1,34 +1,36 @@
+import { Chance } from 'chance';
+const chance = new Chance();
 // Urls for using in the tests
 export const commonUrl = process.env.COMMON_URL || 'https://localhost:5000';
 
 export const ossStandaloneConfig = {
     host: process.env.OSS_STANDALONE_HOST || 'oss-standalone',
     port: process.env.OSS_STANDALONE_PORT || '6379',
-    databaseName: process.env.OSS_STANDALONE_DATABASE_NAME || 'test_standalone',
+    databaseName: process.env.OSS_STANDALONE_DATABASE_NAME + chance.string({ length: 10 }) || 'test_standalone' + chance.string({ length: 10 }),
```
[the remark this answers is not in the document]
Comment (tanyatrayanava) [reply tests/e2e/helpers/conf.ts]: I need to have random db name every time I add new connection, so the random name should be generated each time.

File: tests/e2e/tests/critical-path/browser/context.e2e.ts (hunk 3 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/context.e2e.ts]: done

File: tests/e2e/tests/critical-path/browser/database-overview.e2e.ts (hunk 4 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/database-overview.e2e.ts]: done

File: tests/e2e/tests/critical-path/browser/filtering.e2e.ts (hunk 5 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/filtering.e2e.ts:33]: the same as dbNames, I need to have different keys to remove them after test execution and don't have the same key in other tests.

File: tests/e2e/tests/critical-path/browser/filtering.e2e.ts (hunk 6 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/filtering.e2e.ts:18]: its used in the line 33 and 49

File: tests/e2e/tests/critical-path/browser/hash-field.e2e.ts (hunk 7 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/hash-field.e2e.ts:9]: its used inside tests

File: tests/e2e/tests/critical-path/browser/scan-keys.e2e.ts (hunk 8 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/browser/scan-keys.e2e.ts]: removed

File: tests/e2e/tests/critical-path/cli/cli-critical.e2e.ts (hunk 9 above)
Comment (tanyatrayanava) [reply tests/e2e/tests/critical-path/cli/cli-critical.e2e.ts:23]: its used inside tests

**Review by elenaNaboko — APPROVED** [review]

**Review by ofersteinberg — APPROVED** [review]

**Pull request merged**